src/main/java/graphql/nadel/engine/HydrationInputResolver.java

```diff
@@ -282,12 +324,17 @@ private ExecutionResultNode convertSingleHydrationResultIntoOverallResult(Execut
                                                                               ResultComplexityAggregator resultComplexityAggregator
     ) {
 
-
         Map<String, FieldTransformation> transformationByResultField = queryTransformationResult.getFieldIdToTransformation();
         Map<String, String> typeRenameMappings = queryTransformationResult.getTypeRenameMappings();
+
+        ExecutionResultNode root = rootResultNode.getChildren().get(0);
```

Comment:
lets add an Assert here too to make sure getChildren.size() == 1

Reply:
yep, added